doc/source/serve/deployment-graph/control_flow_based_on_user_inputs.md

```diff
@@ -0,0 +1,52 @@
+# Pattern: Control flow based on the user inputs
+
+The example shows how to user inputs to control the graph flow
+
+![pic](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/deployment-graph/control_flow_based_on_user_inputs.svg)
+
+## Code
+
++++
+
+```{eval-rst}
+.. literalinclude:: ../doc_code/deployment_graph_control_flow_based_on_user_inputs.py
+   :language: python
+```
+
+````{note}
+1. The dag.execute() take arbitrary number of arguments, and internally we implemented data objects to facilitate accessing by index or key.
+
+code example:
+```python
+dag = combine.bind(output1, output2, user_input[1])
+```
```

Comment:
you need to indent two spaces to make it properly nested. otherwise it looks like 
<img width="785" alt="image" src="https://user-images.githubusercontent.com/21118851/170153908-2e62444b-d06a-4b00-9dfe-011968788990.png">